examples/multiselect.rs

```diff
@@ -19,7 +17,7 @@ fn main() {
         "Pineapple",
     ];
 
-    let validator: MultiOptionValidator<&str> = &|a| {
+    let validator = |a: &[ListOption<&&str>]| {
```

Comment:
This one, I spend about 2 or 3 days on, trying to get `rustc` to do some auto-dereference for this type, to have `&[ListOption<&str>]` instead of the double reference.

Would have loved for `rustc` to just take it without any extra type annotations even, as the type should be clear from the usage, and even rust-analyzer figures out the right type.

After several days, I gave up and just accepted this unfortunate type declaration. Simply couldn't find any way of making `rustc` infer the type or drop the second reference.

Reply:
Yeah that's understandable, maybe at some point rustc will improve in this scenario and we'll get the benefits without any changes.